modules/survey_accounts/test/survey_accountsTest.php

```diff
@@ -208,14 +208,77 @@ function testSurveyAccountsWithoutPermission()
         $this->resetPermissions();
     }
     /**
-     * Tests that, when add a survey without visit tag
-     * it should appear the error message
+     * Tests that errors are correctly displayed when the user enters incorrect
+     * or invalid values.
+     * The error messages used in this function come from the file
+     * modules/survey_accounts/php/addsurvey.class.inc
      *
      * @return void
      */
     function testSurveyAccountsAddSurvey()
     {
-        //Visit does not exist for given candidate.
+        $visitLabel        = 'V1';
+        $invalidVisitLabel = '';
+        $instrument        = 'bmi';
+
+        // Ensure that an instrument must be supplied.
+        $this->safeGet($this->url . "/survey_accounts/");
+        $btn = self::$add;
+        $this->webDriver->executescript(
+            "document.querySelector('$btn').click()"
+        );
+        $this->safeFindElement(
+            WebDriverBy::Name("CandID")
+        )->sendKeys("999999");
+        $this->safeFindElement(
+            WebDriverBy::Name("PSCID")
+        )->sendKeys("8889");
+        $this->safeFindElement(
+            WebDriverBy::Name("VL")
+        )->sendKeys($visitLabel);
+        $this->safeFindElement(
+            WebDriverBy::Name("fire_away")
+        )->click();
+        $bodyText =  $this->safeFindElement(
+            WebDriverBy::cssSelector(".error")
+        )->getText();
+        $this->assertContains(
+            "Please choose an instrument.",
+            $bodyText
+        );
+
+        // Ensure visit label exists for a candidate.
+        $this->safeGet($this->url . "/survey_accounts/");
+        $btn = self::$add;
+        $this->webDriver->executescript(
+            "document.querySelector('$btn').click()"
+        );
+        $this->safeFindElement(
+            WebDriverBy::Name("CandID")
+        )->sendKeys("999999");
+        $this->safeFindElement(
+            WebDriverBy::Name("PSCID")
+        )->sendKeys("8889");
+        $this->safeFindElement(
+            WebDriverBy::Name("VL")
+        )->sendKeys($visitLabel);
```

Comment:
Tested on this branch on my own VM:
```suggestion
        $visitField = $this->safeFindElement(
            WebDriverBy::Name("VL"));
        $visitOption = new WebDriverSelect($visitField);
        $visitOption->selectByValue($visitLabel);
```
When you add a survey, the visit label element is a drop-down menu, so I think the original way that this code is written wasn't properly selecting a drop-down menu option. Changing it to this solved it on my VM. I suggest making this change throughout this test whenever you have to set the visit label.

**Note:** If you implement this change, you need to add this line at the top of the file: 
`use Facebook\WebDriver\WebDriverSelect;
`